File: src/game/entity/Block.java
package game.entity;


import game.models.Coordinates;
import game.ui.GameFrame;
import game.ui.GamePanel;
import game.ui.Utility;

public abstract class Block extends Entity {
    public Block(Coordinates coordinates) {
        super(coordinates);
    }

    @Override
    public int getSize() {
        return GamePanel.GRID_SIZE;
    }
}

File: src/game/engine/world/entity/impl/models/Block.kt
package game.engine.world.entity.impl.models

import game.engine.world.geo.Coordinates
import game.engine.ui.panels.game.PitchPanel

abstract class Block : Entity {
    constructor(id: Long) : super(id)
    constructor(coordinates: Coordinates?) : super(coordinates)

    override val size: Int = SIZE

    override fun eliminated() = destroy()

    open fun destroy() = despawnAndNotify()

    override fun onAttackReceived(value: Int) = destroy()

    companion object {
        val SIZE = PitchPanel.GRID_SIZE
    }
}